fonts/extract-glyphs.py

```diff
@@ -0,0 +1,124 @@
+#! /usr/bin/env python3
+
+import os
+from lxml import etree
+
+# Define svg namespace
+svg_ns = "http://www.w3.org/2000/svg"
+
+######################
+#  Helper Functions  #
+######################
+
+# Open file in same directory as script and retrieve its content
+def get_file_content(filepath):
+   __location__ = os.path.realpath(os.path.dirname(__file__))
+   file = open(os.path.join(__location__, filepath), 'r')
+   content = file.read()
+   file.close()
+   return content
+
+# Write content to file with path relative to the script directory
+def write_file_content(filepath, content):
+   __location__ = os.path.realpath(os.path.dirname(__file__))
+   file = open(os.path.join(__location__, filepath), 'w')
+   file.write(content)
+   file.close()
+
+# Retrieve all elements with given tag name from xml
+def get_elements(xml, tag):
+   root = etree.fromstring(bytes(xml, encoding='utf-8'))
+   return root.findall('.//' + tag)  # XPath, recursive
+
+# Retrieve all elements with given tag name from svg
+def get_svg_elements(root, tag):
+   return root.findall('.//{' + svg_ns + '}' + tag)  # XPath, recursive
+
+# Create standard svg root node
+def create_svg_root():
+   root = etree.Element("svg")
+   root.set("xmlns", svg_ns)
+   root.set("version", "1.1")
+   return root
+
+#########################
+#  Parse supported.xsl  #
+#########################
+
+# Retrieve dictionary mapping glyph code to smufl name
+def get_supported_glyph_codes():
+   xml = get_file_content('supported.xsl')
+   glyphs = get_elements(xml, 'glyph')
+   dict = {}
+   for glyph in glyphs:
+      dict[glyph.attrib["glyph-code"]] = glyph.attrib["smufl-name"]
+   return dict
+
+########################
+#  Parse Fontname.svg  #
+########################
+
+# Read svg file which is passed as argument and retrieve information
+def read_svg_file():
+   font_file_name = os.sys.argv[1]
```

Comment:
There is no documentation about what this argument should be, so this will unhelpfully fail if you pass in the wrong argument. Consider using the `argparse` module to provide some built-in help.